Sources/MapboxDirectionsCLI/CodingOperation.swift

```diff
@@ -146,28 +146,52 @@ class CodingOperation<ResponseType : Codable & DirectionsResultsProvider, Option
         return (responseData)
     }
     
-    init(options: ProcessingOptions) {
+    init(options: ProcessingOptions, credentials: Credentials) {
         self.options = options
+        self.credentials = credentials
     }
     
     // MARK: - Command implementation
     
     func execute() throws {
         
-        let config = FileManager.default.contents(atPath: NSString(string: options.configPath).expandingTildeInPath)!
-        let input: Data
-        
-        let decoder = JSONDecoder()
-        
-        let directionsOptions = try decoder.decode(OptionsType.self, from: config)
+        let directions: Directions
+        let directionsOptions: OptionsType
+        let requestURL: URL
+        if FileManager.default.fileExists(atPath: (options.config as NSString).expandingTildeInPath) {
+            // Assume the file is a configuration JSON file. Convert it to an options object.
+            let configData = FileManager.default.contents(atPath: (options.config as NSString).expandingTildeInPath)!
+            let decoder = JSONDecoder()
+            directions = Directions(credentials: credentials)
+            directionsOptions = try decoder.decode(OptionsType.self, from: configData)
+        } else if let url = URL(string: options.config) {
+            // Try to convert the URL to an options object.
+            if let parsedOptions = (RouteOptions(url: url) ?? MatchOptions(url: url)) as? OptionsType {
```

Comment:
This is why `DirectionsOptions(url:)` has to return `nil` when given a URL whose path conflicts with the `abridgedPath` property. If it doesn’t return `nil`, then either RouteOptions and MatchOptions would be equally likely to initialize based on the same URL, mooting the coalescing operator.

Maybe it would be cleaner to nix the `OptionsType` generic parameter in favor of an enumeration with associated values that gets passed into this method, since the two APIs aren’t all that different. But I’ll leave it for now in case another API needs this kind of flexibility.